qiskit/transpiler/passes/calibration/creators.py

```diff
@@ -0,0 +1,472 @@
+
+        if not self._inst_map.has("cx", qubits):
+            raise QiskitError(
+                "This transpilation pass requires the backend to support cx "
+                "between qubits %i and %i." % (q1, q2)
+            )
+
+        cx_sched = self._inst_map.get("cx", qubits=(q1, q2))
+        rzx_theta = Schedule(name="rzx(%.3f)" % theta)
+        rzx_theta.metadata["publisher"] = CalibrationPublisher.QISKIT
+
+        if theta == 0.0:
+            return rzx_theta
+
+        control, target = None, None
+
+        for _, inst in cx_sched.instructions:
+            # Identify the compensation tones.
+            if isinstance(inst.channel, DriveChannel) and isinstance(inst, Play):
+                if isinstance(inst.pulse, GaussianSquare):
+                    target = inst.channel.index
+                    control = q1 if target == q2 else q2
+
+        if control is None:
+            raise QiskitError("Control qubit is None.")
+        if target is None:
+            raise QiskitError("Target qubit is None.")
+
+        if control != qubits[0]:
+            raise QiskitError(
+                "RZXCalibrationBuilderNoEcho only supports hardware-native RZX gates."
+            )
+
+        # Get the filtered Schedule instructions for the CR gates and compensation tones.
+        crs = cx_sched.filter(*[self._filter_control]).instructions
+        rotaries = cx_sched.filter(*[self._filter_drive]).instructions
+
+        # Stretch/compress the CR gates and compensation tones.
+        cr = self.rescale_cr_inst(crs[0][1], 2 * theta)
+        rot = self.rescale_cr_inst(rotaries[0][1], 2 * theta)
+
+        # Build the schedule for the RZXGate without the echos.
+        rzx_theta = rzx_theta.insert(0, cr)
+        rzx_theta = rzx_theta.insert(0, rot)
+        rzx_theta = rzx_theta.insert(0, Delay(cr.duration, DriveChannel(control)))
+
+        return rzx_theta
+
+
+class PulseGates(CalibrationCreator):
+    """Pulse gate adding pass.
+
+    This pass adds gate calibrations in InstructionScheduleMap to a quantum circuit.
+    In QASM3 [1], the gate calibration can be encapsulated as ``defcal`` entry that declares
+    the hardware implementation of ``gate`` in specified grammar, such as OpenPulse.
+
+    This pass checks each DAG circuit node and acquires a corresponding schedule from
+    the instruction schedule map object that may be provided by the target backend.
+    Because this map is a mutable object, end-user can provide configured backend to
+    execute the circuit with customized gate implementations.
+
+    This mapping object returns a schedule with "publisher" metadata which is an integer Enum
+    value representing who created the gate schedule.
+    If the gate schedule is provided by end-users, this pass attaches the schedule to
+    the DAG circuit as a ``defcal`` entry.
+
+    This pass allows users to easily override quantum circuit with custom gate definition
+    without directly dealing with those schedules.
```

Comment:
```suggestion

    This pass checks each DAG circuit node and acquires a corresponding schedule from
    the instruction schedule map object that may be provided by the target backend.
    Because this map is a mutable object, the end-user can provide a configured backend to
    execute the circuit with customized gate implementations.

    This mapping object returns a schedule with "publisher" metadata which is an integer Enum
    value representing who created the gate schedule.
    If the gate schedule is provided by end-users, this pass attaches the schedule to
    the DAG circuit as a calibration.

    This pass allows users to easily override quantum circuit with custom gate definitions
    without directly dealing with those schedules.
```